File: packages/db/schema/enums.ts
import { pgEnum } from "drizzle-orm/pg-core";

export const assetClassEnum = pgEnum("asset_class", [
  "commodity",
  "forex",
  "index",
  "crypto",
  "stock",
]);

export const timeframeEnum = pgEnum("timeframe", ["5m", "15m", "1h"]);

export const strategyStatusEnum = pgEnum("strategy_status", [
  "draft",
  "active",
  "deprecated",
  "archived",
]);

export const riskProfileSlugEnum = pgEnum("risk_profile_slug", [
  "conservative",
  "balanced",
  "aggressive",
]);

export const portfolioModeEnum = pgEnum("portfolio_mode", [
  "backtest",
  "paper",
  "live_manual",
  "live_automated",
]);

export const portfolioStatusEnum = pgEnum("portfolio_status", [
  "active",
  "halted",
  "closed",
]);

export const signalActionEnum = pgEnum("signal_action", [
  "buy",
  "sell",
  "hold",
  "close",
  "modify",
]);

export const decisionTypeEnum = pgEnum("decision_type", [
  "hold",
  "buy_signal",
  "sell_signal",
  "close_signal",
  "no_setup",
  "risk_denied",
  "risk_approved",
  "position_open",
  "trading_halted",
  "execution_failed",
  "sl_triggered",
  "tp_triggered",
  "system_error",
]);

export const directionEnum = pgEnum("direction", ["long", "short"]);

export const entryTypeEnum = pgEnum("entry_type", ["market", "limit"]);

export const orderIntentStatusEnum = pgEnum("order_intent_status", [
  "pending_execution",
  "executed",
  "rejected",
  "expired",
]);

export const orderTypeEnum = pgEnum("order_type", ["market", "limit"]);

export const orderStatusEnum = pgEnum("order_status", [
  "pending",
  "submitted",
  "filled",
  "partial",
  "cancelled",
  "rejected",
]);

export const fillSideEnum = pgEnum("fill_side", ["entry", "exit"]);

export const positionStatusEnum = pgEnum("position_status", [
  "open",
  "closed",
]);

export const exitReasonEnum = pgEnum("exit_reason", [
  "sl",
  "tp",
  "strategy",
  "manual",
  "risk_halt",
  "end_of_backtest",
]);

export const backtestStatusEnum = pgEnum("backtest_status", [
  "pending",
  "running",
  "completed",
  "failed",
]);

export const experimentStatusEnum = pgEnum("experiment_status", [
  "draft",
  "running",
  "completed",
]);

export const jobTypeEnum = pgEnum("job_type", [
  "fetch_data",
  "run_strategy",
  "check_sl_tp",
  "snapshot",
  "backtest",
]);

export const workerJobStatusEnum = pgEnum("worker_job_status", [
  "pending",
  "running",
  "completed",
  "failed",
]);

export const workerRunStatusEnum = pgEnum("worker_run_status", [
  "success",
  "partial",
  "failed",
]);

export const eventSeverityEnum = pgEnum("event_severity", [
  "info",
  "warning",
  "error",
  "critical",
]);

/** Alias for run_context on trading records — same values as portfolio_mode */
export const runModeEnum = portfolioModeEnum;

File: apps/engine/quantara_engine/models/enums.py
"""PostgreSQL enum names used across models."""

import enum


class AssetClass(str, enum.Enum):
    COMMODITY = "commodity"
    FOREX = "forex"
    INDEX = "index"
    CRYPTO = "crypto"
    STOCK = "stock"


class Timeframe(str, enum.Enum):
    M5 = "5m"
    M15 = "15m"
    H1 = "1h"


class StrategyStatus(str, enum.Enum):
    DRAFT = "draft"
    ACTIVE = "active"
    DEPRECATED = "deprecated"
    ARCHIVED = "archived"


class RiskProfileSlug(str, enum.Enum):
    VERY_CONSERVATIVE = "very_conservative"
    CONSERVATIVE = "conservative"
    BALANCED = "balanced"
    AGGRESSIVE = "aggressive"
    VERY_AGGRESSIVE = "very_aggressive"


class PortfolioMode(str, enum.Enum):
    BACKTEST = "backtest"
    PAPER = "paper"
    LIVE_MANUAL = "live_manual"
    LIVE_AUTOMATED = "live_automated"


class PortfolioStatus(str, enum.Enum):
    ACTIVE = "active"
    HALTED = "halted"
    CLOSED = "closed"


class SignalAction(str, enum.Enum):
    BUY = "buy"
    SELL = "sell"
    HOLD = "hold"
    CLOSE = "close"
    MODIFY = "modify"


class DecisionType(str, enum.Enum):
    HOLD = "hold"
    BUY_SIGNAL = "buy_signal"
    SELL_SIGNAL = "sell_signal"
    CLOSE_SIGNAL = "close_signal"
    NO_SETUP = "no_setup"
    RISK_DENIED = "risk_denied"
    RISK_APPROVED = "risk_approved"
    POSITION_OPEN = "position_open"
    TRADING_HALTED = "trading_halted"
    EXECUTION_FAILED = "execution_failed"
    SL_TRIGGERED = "sl_triggered"
    TP_TRIGGERED = "tp_triggered"
    SYSTEM_ERROR = "system_error"


class Direction(str, enum.Enum):
    LONG = "long"
    SHORT = "short"


class EntryType(str, enum.Enum):
    MARKET = "market"
    LIMIT = "limit"


class OrderIntentStatus(str, enum.Enum):
    PENDING_EXECUTION = "pending_execution"
    EXECUTED = "executed"
    REJECTED = "rejected"
    EXPIRED = "expired"


class OrderType(str, enum.Enum):
    MARKET = "market"
    LIMIT = "limit"


class OrderStatus(str, enum.Enum):
    PENDING = "pending"
    SUBMITTED = "submitted"
    FILLED = "filled"
    PARTIAL = "partial"
    CANCELLED = "cancelled"
    REJECTED = "rejected"


class FillSide(str, enum.Enum):
    ENTRY = "entry"
    EXIT = "exit"


class PositionStatus(str, enum.Enum):
    OPEN = "open"
    CLOSED = "closed"


class ExitReason(str, enum.Enum):
    SL = "sl"
    TP = "tp"
    STRATEGY = "strategy"
    MANUAL = "manual"
    RISK_HALT = "risk_halt"
    END_OF_BACKTEST = "end_of_backtest"


class BacktestStatus(str, enum.Enum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"


class ExperimentStatus(str, enum.Enum):
    DRAFT = "draft"
    RUNNING = "running"
    COMPLETED = "completed"


class JobType(str, enum.Enum):
    FETCH_DATA = "fetch_data"
    RUN_STRATEGY = "run_strategy"
    CHECK_SL_TP = "check_sl_tp"
    SNAPSHOT = "snapshot"
    BACKTEST = "backtest"


class WorkerJobStatus(str, enum.Enum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"


class WorkerRunStatus(str, enum.Enum):
    SUCCESS = "success"
    PARTIAL = "partial"
    FAILED = "failed"


class EventSeverity(str, enum.Enum):
    INFO = "info"
    WARNING = "warning"
    ERROR = "error"
    CRITICAL = "critical"
